### PR TITLE
feat: add trigger-release-pr github action

### PR DESCRIPTION
**Pull request by tomdim**
## Description

Add github action for creating automated prs for release to production.

## Checklist

- [X] I have reviewed my code and checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [ ] I have created at least one test case for the changes I have made
- [X] I have updated the documentation (**SwaggerHub**, **README.md**, **Wiki**, etc.) for the changes I have made

### CODE REVIEW

**Review by anmarkoulis — APPROVED** [review]

**Review by vtrihos — APPROVED** [review]

**Comment by dkoutsonikolis** [comment]
Should we maybe create a release branch as well(create a branch from `develop` and merge it to `main`)? I am thinking of cases where `main` might have commits not present in `develop` and possible conflict resolution needed.

File: .github/workflows/trigger-release-pr.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        type: string
+      headBranchName:
+        description: 'Staging/Development branch name (default: `develop`)'
+        required: false
+        default: 'develop'
+        type: string
+      releasePrTitle:
+        description: 'The title of the pull request. (default: `Release to Production {{ today.date }}`)'
+        required: false
+        default: ''
```
[the remark this answers is not in the document]
Comment (tomdim) [reply .github/workflows/trigger-release-pr.yml:18]: {{ today.date }} cannot be evaluated. can't find a way to get the today's date in inputs section. this is why the default value is empty string and handle it inside github script below.

[the remark this answers is not in the document]
Comment (dkoutsonikolis) [reply .github/workflows/trigger-release-pr.yml:18]: Ok great, didn't know that.

**Review by dkoutsonikolis — APPROVED** [review]

File: .github/workflows/trigger-release-pr.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomdim) [reply .github/workflows/trigger-release-pr.yml:18]: I added {{today.date}} as a placeholder. 😄

**Comment by tomdim** [comment]
> Should we maybe create a release branch as well(create a branch from `develop` and merge it to `main`)? I am thinking of cases where `main` might have commits not present in `develop` and possible conflict resolution needed.

I think this maybe better to be done manually. 😰

**Comment by dkoutsonikolis** [comment]
> > Should we maybe create a release branch as well(create a branch from `develop` and merge it to `main`)? I am thinking of cases where `main` might have commits not present in `develop` and possible conflict resolution needed.
> 
> I think this maybe better to be done manually. 😰

The conflict resolution will be done manually (if you mean that) but it might be safer to push them in a release branch instead of directly into `develop`.

**Comment by tomdim** [comment]
> > > Should we maybe create a release branch as well(create a branch from `develop` and merge it to `main`)? I am thinking of cases where `main` might have commits not present in `develop` and possible conflict resolution needed.
> > 
> > 
> > I think this maybe better to be done manually. 😰
> 
> The conflict resolution will be done manually (if you mean that) but it might be safer to push them in a release branch instead of directly into `develop`.

Yes I agree. We could use this github action https://github.com/marketplace/actions/create-pull-request, which pushes to a new branch as well. I could cover this case in a next pr. It could be configurable: create a release branch or immediately merge develop into main.

**Pull request merged**